### ISSUE TITLE
Manual certonly request fails due to newlines in .well-known string

### ISSUE THREAD

**Issue by tedb** [letsencrypt/letsencrypt#1451]
tl;dr: Self-verify of challenge from HTTP server fails; algorithm needs to chomp/trim newlines present in HTTP response.

I am participating in the closed beta and I requested my first cert yesterday.  I am terminating SSL on an external provider so I used the following invocation for letsencrypt-auto:

./letsencrypt-auto -a manual certonly

This process requested that I place a random token at a URL path under /.well-known/acme-challenge/ .  I did so, and tested it with curl, but it still failed to validate.  The error I got was "Self-verify of challenge failed, authorization abandoned."  I ran it again with verbose logging, and noticed this in the log:

2015-11-09 19:11:02,800:DEBUG:acme.challenges:Key authorization from response (u'X6K-MQsdzpMUVNNGoyboRVIxmvfTnNEPbwIoo6Cy2ZU.1WvRR9HsnS_PQRS4VJgCE_gMf8Y28ELCVGXCPlbqVNA
') doesn't match HTTP response (u'X6K-MQsdzpMUVNNGoyboRVIxmvfTnNEPbwIoo6Cy2ZU.1WvRR9HsnS_PQRS4VJgCE_gMf8Y28ELCVGXCPlbqVNA\n\n')

Note the strings are only different because of the newlines.  I generated the file the first time using vi, which added a newline for me.  I had to use "echo -n" to generate the file instead, which wrote the string with no newline.  Many people will embed a newline when writing this file and I think the Let's Encrypt client should be modified to tolerate it.

**Comment by bmw** [comment]
Dupe of #1322.
